src/dos/drive_virtual.cpp

```diff
@@ -374,29 +374,36 @@ bool Virtual_File::Write(uint8_t* /*data*/, uint16_t* /*size*/)
 
 bool Virtual_File::Seek(uint32_t* new_pos, uint32_t type)
 {
+	uint32_t requested_pos = {};
+
 	switch (type) {
 	case DOS_SEEK_SET:
 		if (*new_pos <= file_data->size()) {
-			file_pos = *new_pos;
+			requested_pos = *new_pos;
 		} else {
 			return false;
 		}
 		break;
 	case DOS_SEEK_CUR:
 		if ((*new_pos + file_pos) <= file_data->size()) {
-			file_pos = *new_pos + file_pos;
+			requested_pos = *new_pos + file_pos;
 		} else {
 			return false;
 		}
 		break;
 	case DOS_SEEK_END:
 		if (*new_pos <= file_data->size()) {
-			file_pos = file_data->size() - *new_pos;
+			requested_pos = file_data->size() - *new_pos;
 		} else {
 			return false;
 		}
 		break;
 	}
+
+	if (file_pos != requested_pos) {
+		file_pos = requested_pos;
+	}
```

Comment:
Remove the if satement.  Assuming these are normal integers and there's no weird copy constructor behavior, it doesn't accomplish anything to compare them before setting them equal.

Reply:
I tend to agree; I don't see why this is necessary. Old code was fine.

Reply:
Yep sorry I missed that.  If statement makes sense then.

Now that I think about it though, should we really be adding a delay to Seek at all?  It doesn't necessarily move the drive head.  MS-DOS Encyclopedia has the interrupt documented with the name "Move File Pointer" rather than "Seek" because really all it needs to do is update an in-memory internal variable to point to the next byte it wants to read/write.  It shouldn't need to actually access the disk until the next read or write.

Reply:
You're right, most likely it works like the C seek call; it only updates the in-memory file pointer which will then be take in to consideration on the next read/write call.